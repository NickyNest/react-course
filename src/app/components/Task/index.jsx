import React from 'react';
import TaskTag from 'components/TaskTag';
import {Checkbox} from 'semantic-ui-react';
import {shape, string, bool, instanceOf, func} from 'prop-types';
import './index.css';

const Task = ({task, onCheckChange, onRemoveTask}) => (
    <div>
        <Checkbox toggle checked={task.completed} onChange={(e, {checked}) => onCheckChange(task.id, checked)} />
        <span className={task.completed ? 'throw-text' : ''}>{task.title}</span>&nbsp;&nbsp;
        <span>{task.createdDate.toLocaleDateString()}</span>
        {task.tags.length === 0 ? '' : task.tags.map(tag => <TaskTag key={tag.tagId} taskTag={tag} />)}
        <button onClick={() => onRemoveTask(task.id)}>X</button>
    </div>
);

Task.propTypes = {
    task: shape({
        id: string.isRequired,
        title: string.isRequired,
        completed: bool.isRequired,
        createdDate: instanceOf(Date).isRequired
    }).isRequired,
    onCheckChange: func.isRequired,
    onRemoveTask: func.isRequired
};

export default Task;