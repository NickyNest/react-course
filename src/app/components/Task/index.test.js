/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Task from './index';

configure({adapter: new Adapter()});

describe('Task', () => {
    it('Task (completed=true) match expected snapshot', () => {
        expect(shallow(<Task
            task={{id: '1', title: 'temp', completed: true, createdDate: new Date(1900, 1, 1)}}
            onCheckChange={() => {}}
            onRemoveTask={() => {}} />)).toMatchSnapshot();
    });
});

describe('Task', () => {
    it('Task (completed=false) match expected snapshot', () => {
        expect(shallow(<Task
            task={{id: '1', title: 'temp', completed: false, createdDate: new Date(1900, 1, 1)}}
            onCheckChange={() => {}}
            onRemoveTask={() => {}} />)).toMatchSnapshot();
    });
});

