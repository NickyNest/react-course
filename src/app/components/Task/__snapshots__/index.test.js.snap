// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Task (completed=false) match expected snapshot 1`] = `
<div>
  <Checkbox
    checked={false}
    onChange={[Function]}
    toggle={true}
    type="checkbox"
  />
  <span
    className=""
  >
    temp
  </span>
    
  <span>
    2/1/1900
  </span>
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;

exports[`Task Task (completed=true) match expected snapshot 1`] = `
<div>
  <Checkbox
    checked={true}
    onChange={[Function]}
    toggle={true}
    type="checkbox"
  />
  <span
    className="throw-text"
  >
    temp
  </span>
    
  <span>
    2/1/1900
  </span>
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
